public/app/features/datasources/settings/DataSourceSettingsPage.tsx

```diff
@@ -230,11 +231,12 @@ export class DataSourceSettingsPage extends PureComponent<Props> {
 
   renderSettings() {
     const { dataSourceMeta, setDataSourceName, setIsDefault, dataSource, plugin, testingStatus } = this.props;
-    const canEditDataSources = contextSrv.hasPermission(AccessControlAction.DataSourcesWrite);
+    const canWriteDataSource = contextSrv.hasPermissionInMetadata(AccessControlAction.DataSourcesWrite, dataSource);
```

Comment:
I'm a bit confused here. So if I don't have the `accesscontrol` feature enabled, will I always be able to edit the datasource? Regardless of my permissions?

Reply:
Good question. Yes, you'd always be able to edit the data-source, but without the `accesscontrol` feature enabled, access control is based on the user role. Hence you'd only have access to this page if you are an `orgAdmin`, which are allowed to.

Reply:
Ok ok, cool!